modules/build/src/main/scala/scala/build/internal/AmmUtil.scala

```diff
@@ -13,7 +13,7 @@ object AmmUtil {
       case i  => fileName.take(i)
     }
 
-    (pkg, Name(wrapper))
+    (pkg, Name(s"${wrapper}_scalacli_wrapper"))
```

Comment:
We were also thinking about adding a random suffix, but that would make it impossible to run the main class by name. So instead we did a named that should not interfere with any library. This might be breaking so we also wanted to have a fallback to add the suffix if someone is using the old one. @Gedochao any idea where that can be?

Reply:
Sorry for getting back late.
As for inferring main classes, detecting script main classes and such, the logic is here:
https://github.com/VirtusLab/scala-cli/blob/a4c286ff026a0a77142d2ae226b6a3fd1fb6f692/modules/build/src/main/scala/scala/build/Build.scala#L115-L183

Reply:
However, I am not convinced we should change the wrapper name - not without explicit configuration, at least.
Maybe via directive/flag.
Changing the default behaviour will be an effective change of syntax, not just for inferring the main class name.
i.e. code like this will stop working:
https://gist.github.com/Gedochao/8902375ad97d24d8ed49c9bb33d5bd08
And this feels quite bad to me, since we've had this syntax since forever.

Reply:
Maybe something like:
```scala
//> using script.overrideWrapperName
```
?
So that the change of wrapper is committed in the code, to make the changed behaviour predictable.

Reply:
As for better UX, we could think about detecting the cases such as the reproduction from https://github.com/VirtusLab/scala-cli/issues/3595, and then print a warning, suggesting to override the wrapper... Although detecting such a case would likely mean parsing AST, so that's a tougher cookie.